### PR TITLE
feat(DoDontBlock): add internal link chooser to RTE

### PR DESCRIPTION
**Pull request by fulopdaniel**
- Reuse the RTE from shared folder
- Add custom plugins to RTE

### CODE REVIEW

File: packages/dos-donts-block/src/DoDontItem.tsx [hunk 1]
```diff
@@ -155,16 +150,15 @@ export const DoDontItem = React.forwardRef<HTMLDivElement, DoDontItemProps>(
         const memoizedRichTextEditor = useMemo(
```
Comment (ragi96) [inline packages/dos-donts-block/src/DoDontItem.tsx:150]: Is the memoization still needed?

File: packages/dos-donts-block/src/helpers/getPlugins.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    ResetFormattingPlugin,
+    SoftBreakPlugin,
+    StrikethroughPlugin,
+    TextStylePlugin,
+    UnderlinePlugin,
+    UnorderedListPlugin,
+} from '@frontify/fondue';
+import { ButtonPlugin, LinkPlugin } from '@frontify/guideline-blocks-shared';
+
+export const getPlugins = (appBridge: AppBridgeBlock) => {
```
Comment (ragi96) [inline packages/dos-donts-block/src/helpers/getPlugins.ts:26]: Probably it makes sense to but some (Titile, description [this one?]) Plugin Settings into the shared, so we can reuse them everywhere

File: packages/dos-donts-block/src/helpers/getPlugins.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                new AlignRightPlugin(),
+                new AlignJustifyPlugin(),
+                new UnorderedListPlugin(),
+                new CheckboxListPlugin(),
+                new OrderedListPlugin(),
+                new ResetFormattingPlugin(),
+                new EmojiPlugin(),
+            ]
+        );
+    return plugins;
```
Comment (ragi96) [inline packages/dos-donts-block/src/helpers/getPlugins.ts:53]:
```suggestion
    return = new PluginComposer()
        .setPlugin(new SoftBreakPlugin(), new ParagraphPlugin())
        .setPlugin(new TextStylePlugin())
        .setPlugin(
            [
                new BoldPlugin(),
                new ItalicPlugin(),
                new UnderlinePlugin(),
                new StrikethroughPlugin(),
                new LinkPlugin({ appBridge }),
                new ButtonPlugin({ appBridge }),
                new CodePlugin(),
            ],
            [
                new AlignLeftPlugin(),
                new AlignCenterPlugin(),
                new AlignRightPlugin(),
                new AlignJustifyPlugin(),
                new UnorderedListPlugin(),
                new CheckboxListPlugin(),
                new OrderedListPlugin(),
                new ResetFormattingPlugin(),
                new EmojiPlugin(),
            ]
        );
```

File: packages/dos-donts-block/src/DoDontItem.tsx (hunk 1 above)
Comment (fulopdaniel) [reply packages/dos-donts-block/src/DoDontItem.tsx:150]: Yes unfortunately: Drag and dropping rerenders the component a LOT of times, and if we need to rerender the RTE each time it becomes a little laggy.

File: packages/dos-donts-block/src/helpers/getPlugins.ts (hunk 2 above)
Comment (fulopdaniel) [reply packages/dos-donts-block/src/helpers/getPlugins.ts]: Created a `pluginPresets` library in the shared folder. What do you think?

File: packages/dos-donts-block/src/helpers/getPlugins.ts (hunk 3 above)
Comment (fulopdaniel) [reply packages/dos-donts-block/src/helpers/getPlugins.ts]: thanks, done!

File: packages/dos-donts-block/src/DoDontItem.tsx (hunk 1 above)
Comment (ragi96) [reply packages/dos-donts-block/src/DoDontItem.tsx:155]: Ok then its fine, i just thought with the new handling that maybe also improved

File: packages/dos-donts-block/src/helpers/getPlugins.ts (hunk 2 above)
Comment (ragi96) [reply packages/dos-donts-block/src/helpers/getPlugins.ts]: Cool! Thanks.

**Review by ragi96 — APPROVED** [review]

**Review by celine-s — APPROVED** [review]
nice!

**Pull request merged**